util/test/chpl_launchcmd.py

```diff
@@ -1074,13 +1074,17 @@ def submit_job(self, testing_dir, output_file, error_file, input_file):
             logging.error(msg)
             raise ValueError(msg)
 
-        # Output is: Submitted batch job 106001
-        id_parts = stdout.split(' ')
-        if len(id_parts) < 4:
-            raise ValueError('Could not parse output from sbatch submission: {0}'.format(stdout))
-        else:
-            job_id = id_parts[3].strip()
-            return job_id
+        # The output line we want is: Submitted batch job 106001
+        lines = stdout.split('\n')
+        for line in lines:
+            if line.startswith('Submitted batch job'):
+                id_parts = line.split(' ')
+                if len(id_parts) < 4:
+                    raise ValueError('Could not parse output from sbatch submission: {0}'.format(stdout))
+                else:
+                    job_id = id_parts[3].strip()
+                    return job_id
+        raise ValueError('Could not parse output from sbatch submission: {0}'.format(stdout))
```

Comment:
Isn't this supposed to be in an else block for "Submitted batch job" check?

Reply:
No, it's the else-block for `if len...`.

Reply:
I misunderstood the question, which led to mis-answering it.  The question really had to do with the final `raise`.  That's in the right place; it's the backstop if we don't match any lines at all.  The follow-up commit makes this clearer.